src/dsl.jl

```diff
@@ -688,25 +692,41 @@ function read_equations_options(options, variables_declared)
             error("Malformed equation: \"$eq\". Equation's left hand and right hand sides should be separated by a \"~\".")
         end
 
-        # Checks if the equation have the format D(X) ~ ... (where X is a symbol). This means that the
-        # default differential has been used. X is added as a declared variable to the system, and
-        # we make a note that a differential D = Differential(iv) should be made as well.
```

Comment:
seems some stuff here got removed

Reply:
It was refactored to a function that looks like it is trying to handle more general LHS. All of that makes sense to do as we should try to support any valid ODE that MTK can support, but that should really be a in a followup PR to generalize what we support (keeping this focused on just fixing the function bug).